Setting property naming strategy without any data classes causes build failure

Thanks for reporting an issue, please review the task list below before submitting the
issue. Your issue report will be closed if the issue is incomplete and the below tasks not completed.

NOTE: If you are unsure about something and the issue is more of a question a better place to ask questions is on Stack Overflow (https://stackoverflow.com/tags/micronaut) or Gitter (https://gitter.im/micronautfw/). DO NOT use the issue tracker to ask questions.

### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Generate new project via launch and only select openapi
2. Add a single controller that returns a built-in class like string or map
3. Set the property naming strategy to something not the default (like `SNAKE_CASE`)

### Expected Behaviour

Build would pass and a openapi yml file would be generated.

### Actual Behaviour

The gradle build fails and no openapi yml is generated

### Environment Information

- **Operating System**: macOS 11.0.1 and 10.15.6
- **Micronaut Version:** 2.3.0
- **JDK Version:** 11

### Example Application

[Example Repository](https://github.com/tmapes/openapidemo)

#### Output from Gradle Build
```
11:15:24 AM: Executing tasks ':classes :testClasses'...

> Task :kaptGenerateStubsKotlin
> Task :processResources
> Task :processTestResources NO-SOURCE

> Task :kaptKotlin FAILED
Note: Generating OpenAPI DocumentationNote: Using SNAKE_CASE property naming strategy.Note: Creating bean classes for 1 type elementsNote: Using SNAKE_CASE property naming strategy.3 actionable tasks: 3 executed
error: Error finalizing type visitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor@68f5ee16]: nullerror: Error finalizing type visitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor@68f5ee16]: null
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kaptKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.gradle.internal.KaptExecution
   > java.lang.reflect.InvocationTargetException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
11:15:27 AM: Tasks execution finished ':classes :testClasses'.

```

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java
@@ -409,22 +409,24 @@ public class OpenApiApplicationVisitor extends AbstractOpenApiVisitor implements
         final PropertyNamingStrategies.NamingBase propertyNamingStrategy = fromName(namingStrategyName);
         if (propertyNamingStrategy != null) {
             visitorContext.info("Using " + namingStrategyName + " property naming strategy.");
-            openAPI.getComponents().getSchemas().values().forEach(model -> {
-                Map<String, Schema> properties = model.getProperties();
-                if (properties != null) {
-                    Map<String, Schema> newProperties = properties.entrySet().stream()
+            if (openAPI.getComponents() != null && CollectionUtils.isNotEmpty(openAPI.getComponents().getSchemas())) {
+                openAPI.getComponents().getSchemas().values().forEach(model -> {
+                    Map<String, Schema> properties = model.getProperties();
+                    if (properties != null) {
+                        Map<String, Schema> newProperties = properties.entrySet().stream()
                             .collect(Collectors.toMap(entry -> propertyNamingStrategy.translate(entry.getKey()),
-                                    Map.Entry::getValue, (prop1, prop2) -> prop1, LinkedHashMap::new));
-                    model.getProperties().clear();
-                    model.setProperties(newProperties);
-                }
-                List<String> required = model.getRequired();
-                if (required != null) {
-                    List<String> updatedRequired = required.stream().map(propertyNamingStrategy::translate).collect(Collectors.toList());
-                    required.clear();
-                    required.addAll(updatedRequired);
-                }
-            });
+                                Map.Entry::getValue, (prop1, prop2) -> prop1, LinkedHashMap::new));
+                        model.getProperties().clear();
+                        model.setProperties(newProperties);
+                    }
+                    List<String> required = model.getRequired();
+                    if (required != null) {
+                        List<String> updatedRequired = required.stream().map(propertyNamingStrategy::translate).collect(Collectors.toList());
+                        required.clear();
+                        required.addAll(updatedRequired);
+                    }
+                });
+            }
         }
     }
 

